### PR TITLE
Fix various compiler warnings

### PR DESCRIPTION
**Pull request by holden-zenith**
This fixes various compiler warnings found when testing with stm32f7

### CODE REVIEW

File: source/include/FreeRTOSIPConfigDefaults.h
Comment (AniruddhaKanhere) [inline source/include/FreeRTOSIPConfigDefaults.h:255]: The `#warning` option is not supported on windows simulator.

Since this file is not specific to STM, I think we should not make this change...

File: source/include/FreeRTOSIPConfigDefaults.h
Comment (AniruddhaKanhere) [inline source/include/FreeRTOSIPConfigDefaults.h:279]: Same here.

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]
Thank you @holden-zenith for this PR.
I have one suggested change, can you please take a look?

**Review by htibosch — APPROVED** [review]
Turn the warning into an error, I'm OK with that change.
Thank you

**EDIT**
Sorry I pressed on a link from my email client and I only saw part of your PR. I will read (and review) more.

**Review by AniruddhaKanhere — APPROVED** [review]
Thanks for this PR @holden-zenith. I am approving this.

We'll need one more review - I am asking for it. Once we have that, we can merge the PR!

**Review by paulbartell — APPROVED** [review]

**Pull request merged**
